parsers/mcmaster-parser-2.js

```diff
@@ -0,0 +1,125 @@
+    var _sectionTypeRegex = /(\d+)\s+([A-Z]{1}\d+)\s+(?=Lecture|Tutorial|Laboratory)([A-Za-z]{3,}\b)/;
+    var _dayRegex = /(Mo)|(Tu)|(We)|(Th)|(Fr)|(Sa)|(Su)/g;
+    var _weekdays = {"Mo":"Monday","Tu":"Tuesday","We":"Wednesday","Th":"Thursday","Fr":"Friday","Sa":"Saturday","Su":"Sunday"};
+
+    function convertTo24Hour(time) {
+        var hours = parseInt(time.substr(0, 2));
+        if(time.indexOf('AM') != -1 && hours == 12) {
+            time = time.replace('12', '0');
+        }
+        if(time.indexOf('PM')  != -1 && hours < 12) {
+            time = time.replace(hours, (hours + 12));
+        }
+        return time.replace(/(AM|PM)/, '');
+    }
+
+    var timetable = [];
+    var courses = text.split(_courseTitleLookAhead);
+    for (var i = 0; i < courses.length; i++) {
+        var courseText = courses[i];
+        var course = _courseTitleRegex.exec(courseText);
+        var classes = courseText.split(_timeDayPairOrCourseCodeLookAhead);
+        if (!course) continue;
+        var currentSection = _sectionTypeRegex.exec(courseText);
+        //console.log(classes);
+
+        for (var j = 0; j < classes.length; j++) {
+            var locationProfPair = _locationProfPair.exec(classes[j]);
+            var time = _timeDayPairRegex.exec(classes[j]);
+            var semester = _semesterPairRegex.exec(classes[j]);
+            var section = _sectionTypeRegex.exec(classes[j]);
+            currentSection = section ? section : currentSection; // Check is the current classes segment is a section if it is update the current Section
+
+            if (!locationProfPair || !time || !semester) continue;
+
+            var where = locationProfPair[1];
+            var professor = locationProfPair[2];
+            var days = time[1].replace(_dayRegex, function myFunction(x){return _weekdays[x] + ",";});
+            var daysArray = days.substring(0, days.length-1).split(",");
+
+            for (var k = 0; k < daysArray.length ; k++) {
+                var obj = {};
+                var startTimeSplit = convertTo24Hour(time[2]).split(":");
+                var startTimeObj = {'hour': parseInt(startTimeSplit[0], 10), 'minute': parseInt(startTimeSplit[1], 10)};
+                var endTimeSplit = convertTo24Hour(time[3]).split(":");
+                var endTimeObj = {'hour': parseInt(endTimeSplit[0], 10), 'minute': parseInt(endTimeSplit[1], 10)};
+                var sem_start = new Date(semester[1]);
+                var sem_end = new Date(semester[2]);
+
+                var semesterStartObj = { year: sem_start.getFullYear(), month: sem_start.getMonth(), day: sem_start.getDate()};
+                var semesterEndObj = { year: sem_end.getFullYear(), month: sem_end.getMonth(), day: sem_end.getDate()};
+                var courseSplit = course[1].split(" ");
+                obj['course_code_faculty'] = courseSplit[0];
```

Comment:
I'll do the same for my parse, can you make this so that it's like this instead:

```
timetable.push({
   'course_code_number': courseSplit[1],
   'course_number': courseSection[1],
   etc etc
})
```

Easier to read and don't have to construct a temporary obj.